src/lib/adamantServerApi.js

```diff
@@ -519,6 +520,216 @@ function install (Vue) {
+    let keys = this.getKeypair()
+    let transaction = {
+      type: constants.Transactions.VOTE,
+      asset: {votes: votes},
+      timestamp: adamant.epochTime(),
+      recipientId: this.$store.state.address,
+      senderPublicKey: keys.publicKey.toString('hex'),
+      amount: 0,
+      senderId: this.$store.state.address
+    }
+    transaction.signature = adamant.transactionSign(transaction, keys)
+    this.$store.commit('clean_delegates')
+    this.$store.commit('ajax_start')
+    this.$http.post(this.getAddressString() + '/api/accounts/delegates', transaction).then(response => {
+      if (response.body.success) {
+        this.$store.commit('set_last_transaction_status', false)
+        // removing an UI waiting state if transaction confirmation run to much time
+        window.setTimeout(() => {
+          if (!this.$store.state.lastTransactionConfirmed) {
+            this.$store.commit('send_error', { msg: this.$t('votes.transaction_confirm_await') })
+          }
+          this.$store.commit('ajax_end')
+          this.getDelegatesWithVotes()
+        }, 15000)
+        this.checkUnconfirmedTransactions()
+      } else {
+        this.$store.commit('send_error', { msg: `${this.$t('error')}: ${response.body.error}` })
+        this.getDelegatesWithVotes()
+      }
+    }, () => {
+      this.$store.commit('ajax_end_with_error')
+    })
+  }
+
+  Vue.prototype.getForgingTimeForDelegate = function (delegate) {
+    const getRoundDelegates = (delegates, height) => {
+      let currentRound = round(height)
+      return delegates.filter((delegate, index) => {
+        return currentRound === round(height + index + 1)
+      })
+    }
+    const round = height => {
+      if (isNaN(height)) {
+        return 0
+      } else {
+        return Math.floor(height / 101) + (height % 101 > 0 ? 1 : 0)
+      }
+    }
+
+    this.$store.commit('ajax_start')
+    this.$http.get(this.getAddressString() + '/api/delegates/getNextForgers?limit=101').then(response => {
+      if (response.body.success) {
+        const nextForgers = response.body.delegates
+        const fIndex = nextForgers.indexOf(delegate.publicKey)
+        const forgingTime = fIndex === -1 ? -1 : fIndex * 10
+        this.$store.commit('update_delegate', {
+          address: delegate.address,
+          params: { forgingTime: forgingTime }
+        })
+        this.$http.get(this.getAddressString() + '/api/blocks?orderBy=height:desc&limit=100').then(response => {
+          if (response.body.success) {
+            let lastBlock = response.body.blocks[0]
+            let blocks = response.body.blocks.filter(x => x.generatorPublicKey === delegate.publicKey)
+            let time = Date.now()
+            let status = { updatedAt: time }
+            if (blocks.length > 0) {
+              status.lastBlock = blocks[0]
+              status.blockAt = new Date((((Date.UTC(2017, 8, 2, 17, 0, 0, 0) / 1000) + status.lastBlock.timestamp) * 1000))
```

Comment:
Please, use relevant constants from `constants.js` instead of `Date.UTC(2017, 8, 2, 17, 0, 0, 0)`